### DISCUSSION TITLE
Floating point precision

### DISCUSSION THREAD

**Discussion by niklas-heer** [janet-lang/janet#1158] [Q&A] [answered]
Hi everyone 👋

Sorry, I pretty new to Janet and have been enjoying using it so far, but I came across an issue:
How can I print/use the full precision of IEEE 754 double-precision (mentioned in #446)?

I'm playing around with Janet and want to calculate Pi via the Leibniz formular.
My implementation looks like this:
```janet
(defn calc-pi-leibniz
  "Translation of Java/Clojure solution to Janet"
  [rounds]
  (var pi 1.0)
  (var x 1.0)
  (loop [i :range [2 (+ 2 rounds)]]
            (*= x -1)
            (+= pi (/ x (dec (* 2 i)))))
  (* 4 pi))

(def rounds 100000000)
(print (string (calc-pi-leibniz rounds)))
```
Output: `3.14159`

If I run similar Clojure code, I get much higher precision: `3.141592663589326`
```clojure
(ns leibniz)

(set! *unchecked-math* :warn-on-boxed)

(defn calc-pi-leibniz
  "Translation of Java solution to Clojure"
  [^long rounds]
  (let [end (+ 2 rounds)]
    (loop [i 2 x 1.0 pi 1.0]
      (if (= i end)
        (* 4 pi)
        (let [x (- x)]
          (recur (inc i) x (+ pi (/ x (dec (* 2 i))))))))))

(defn parse-int [s]
  (Integer/parseInt (re-find #"\A-?\d+" s)))

(def rounds 100000000)
(println (calc-pi-leibniz rounds))
```

Would it be possible to get higher precision, similar to Clojure with Janet? If yes, how can I do that?

**Comment by zevv** [comment, marked as answer]
The lack of precision is only in the default representation of floating point numbers. Try this:
```
(printf "%.12f" (calc-pi-leibniz rounds))      
```

**Reply by niklas-heer** [reply]
@zevv thank you so much for the swift response. That totally solves it! 👍 
I had the feeling the precision was there, but just had to be printed correctly. 😅
